crates/solidity/testing/snapshots/cst_output/UsingDeconstructionSymbol/single_id_as_operator/generated/0.8.13-failure.yml

```diff
@@ -6,15 +6,15 @@ Source: >
 Errors: # 1 total
   - >
     Error: Expected Period.
-       ╭─[crates/solidity/testing/snapshots/cst_output/UsingDeconstructionSymbol/single_id_as_operator/input.sol:1:4]
+       ╭─[crates/solidity/testing/snapshots/cst_output/UsingDeconstructionSymbol/single_id_as_operator/input.sol:1:5]
        │
      1 │ foo as /
-       │    ──┬──  
-       │      ╰──── Error occurred here.
+       │     ──┬─  
+       │       ╰─── Error occurred here.
     ───╯
 
 Tree:
   - UsingDeconstructionSymbol: # 0..8 "foo as /"
       - (name) IdentifierPath: # 0..3 "foo"
           - (item) Identifier: "foo" # 0..3
-      - SKIPPED: " as /" # 3..8
+      - SKIPPED: "as /" # 4..8
```

Comment:
not sure I understand what happened to the space here?

Reply:
The top-level `UsingDeconstructionSymbol` was fully parsed and matched `foo`, after which we tried to collect the leading trivia, thus including the whitespace at the end. Then, we SKIPPED over the remaining, unexpected input.

Would you like to handle it differently here?

Reply:
I see. That sounds correct to me, but in that case, the error message should be about the unexpected input `as /` at the end, right?
`Expected Period` here sounds like it is still trying to parse `IdentifierPath`, and failing, so it skipped the remaining input.

Reply:
I think this might be due to the fact that we reuse the previous `expected_tokens` from the successful match? should we be reporting that we expected "nothing" instead?

Reply:
It is, however most of the time it's useful to be able to report the tokens that would allow for progress, see https://github.com/NomicFoundation/slang/commit/709dce892a436b514162623f38f92baad09b53e3 for an example.

If we replace that with "expected end of file", I'd say we regress the error reporting diagnostics in few cases.